sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CallMethodViaReflection.scala

```diff
@@ -229,10 +234,13 @@ object CallMethodViaReflection {
       } else {
         // Argument type must match. That is, either the method's argument type matches one of the
         // acceptable types defined in typeMapping, or it is a super type of the acceptable types.
-        candidateTypes.zip(argTypes).forall { case (candidateType, argType) =>
+        candidateTypes.zip(argTypes).forall { case (candidateType, argType)
+          if !argType.isInstanceOf[StringType] =>
           typeMapping(argType).exists(candidateType.isAssignableFrom)
+        case _ => true
```

Comment:
```suggestion
        case _ => candidateType.isAssignableFrom(classOf[String])
```
I believe this should not be only true. This should be reproducible from passing something that is a string to a function that expects e.g. integer.

Reply:
Also add test for this case